lib/ex_unit/lib/ex_unit/callbacks.ex

```diff
@@ -285,6 +285,43 @@ defmodule ExUnit.Callbacks do
   end
 
   @doc """
+  Same as `start_supervised/2` but returns the PID on success and raises if
+  not started porperly.
+  """
+  @spec start_supervised!(Supervisor.child_spec | module | {module, term}, keyword) :: pid
+  def start_supervised!(child_spec_or_module, opts \\ []) do
+    case start_supervised(child_spec_or_module, opts) do
+      {:ok, pid} ->
+        pid
+      {:ok, pid, _info} ->
+        pid
+      {:error, reason} ->
+        raise """
+        Failed to start child:
+
+        #{format_start_error_reason(reason)}
+        """
+    end
+  end
+
+  defp format_start_error_reason({{:EXIT, reason}, info}),
+    do: format_start_error_reason(reason, info)
+  defp format_start_error_reason({reason, info}),
+    do: format_start_error_reason(reason, info)
+  defp format_start_error_reason(reason),
+    do: Exception.format_exit(reason)
+
+  defp format_start_error_reason(reason, info) do
+    inspect_info(info) <> Exception.format_exit(reason)
+  end
+
+  defp inspect_info({_, _, _, {m, f, a}, _, _, _, _})
+       when is_atom(m) and is_atom(f) and (is_integer(a) or is_list(a)),
```

Comment:
:sweat: we shouldn't be matching on an undocumented record because it could change. It might be that we can ignore the child info. Likely handling bad child specs might change how this function works anyway.

Reply:
I thought this was the child_spec... but you are right. It seems to be an internal supervisor record. @kelvinst let's just ignore this information.

Reply:
Cool! :)